inductiva/scenarios.py

```diff
@@ -2,48 +2,68 @@
 
 from abc import ABC
 import tempfile
-from typing import Optional
+from typing import Optional, List
 
 from inductiva.types import Path
-from inductiva.simulation import Simulator
+from inductiva.simulation import Simulator, Command
 from inductiva.utils.files import resolve_path, get_timestamped_path
 from inductiva.utils.misc import split_camel_case
 
 
 class Scenario(ABC):
     """Base class for scenarios."""
 
-    def simulate(
-        self,
-        simulator: Simulator,
-        output_dir: Optional[Path] = None,
-        **kwargs,
-    ):
-        """Simulates the scenario."""
+    def setup_output_dir(self, output_dir: str):
+        """Setup the scenario output directory."""
         if output_dir is None:
             scenario_name_splitted = split_camel_case(self.__class__.__name__)
             output_dir_prefix = "-".join(scenario_name_splitted).lower()
             output_dir = get_timestamped_path(f"{output_dir_prefix}-output")
-
         output_dir = resolve_path(output_dir)
+        return output_dir
 
-        with tempfile.TemporaryDirectory() as input_dir:
-
-            self.gen_aux_files(simulator, input_dir)
-            self.gen_config(simulator, input_dir)
+    def setup_config(self, simulator: Simulator, input_dir: Path):
```

Comment:
Also here: you can't ask the user of a Scenario to setup configuration files of simulators... that needs to be hidden....

Unless you want to have some private methods (in Python nothing is really private, but you can use the standard of prepending underscore to the method name to say this an internal method to the class, see : https://google.github.io/styleguide/pyguide.html#3162-naming-conventions) and the derived classes for specific Scenarios would use those methods for some auxiliriary thing, but the user of the scenario never needs to use this setup_config methods or simulate_pipeline... that's too low-level to be exposed in a Scenario...